wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1681,7 +1681,61 @@ public final class MLSService: MLSServiceInterface {
     // MARK: - Proteus to MLS Migration
 
     public func startProteusToMLSMigration() async throws {
-        // Migrate proteus team group conversations to MLS
+        guard let context = context else {
+            assertionFailure("MLSService.context is nil")
+            return
+        }
+
+        let groupConversations = try await context.perform {
+            try ZMConversation.fetchAllTeamGroupConversations(messageProtocol: .proteus, in: context)
+                .map {(
+                    $0.qualifiedID,
+                    $0.mlsGroupID,
+                    $0.localParticipants.map { MLSUser(from: $0) }
+                )}
+        }
+        for (qualifiedID, mlsGroupID, members) in groupConversations {
+
+            guard let qualifiedID else {
+                assertionFailure("the group conversation has no `qualifiedID` set")
+                continue
+            }
+            guard let mlsGroupID else {
+                assertionFailure("the group conversation has no `mlsGroupID` set")
+                continue
+            }
```

Comment:
Using `assertionFailure`s here seems quite harsh. Can't we log the error and continue?

Reply:
is it expected to have conv at this point without mlsGroupID or qualifiedId?

Reply:
I restructured the code.